### DISCUSSION TITLE
Mogrify when using DLL - also delete?

### DISCUSSION THREAD

**Discussion by FrontierDK** [ImageMagick/ImageMagick#2687] [General]
Hi all 👍 
I have just tested the mogrify option in the DLL version - and it works really fine. However, when also using the -convert option, the original files aren't deleted.

Is there any flag one can add, which will make the batch mogrify also delete the original files after converting?

**Comment by snibgo** [comment]
There is no "-convert" option. Please show your exact command.

"magick mogrify" will overwrite the input files, unless the "-path" option is used. "-path" will write output files to that path, leaving the input files untouched.

**Reply by FrontierDK** [reply]
my command is as follows (in VBscript):
' --------------------------
option explicit
dim MyObject
set MyObject = CreateObject("ImageMagickObject.MagickImage.1")
MyObject.mogrify "-format", "webp", "-quality", "60", "-define", "webp:method=6", "-define", "webp:filter-strength=0", "*.png"
set MyObject=nothing

It works - only, as the manual says, the original file isn't overwritten when changing the format. Would be really nice if it had a flag/option to delete the old files...

**Comment by snibgo** [comment]
Yes, true, if you use "-format", that gives a new extension for the output file, so the input will not be overwritten.

IM currently has no facility for deleting files. At one time it did, the "ephemeral:" pseudo-format, but that was removed. I'm not sure why.

**Comment by fmw42** [comment]
Write your images to a pre-created new directory and then just delete the old directory.
